sig-apps/Charter.md

```diff
@@ -0,0 +1,141 @@
+# SIG Apps Charter
+
+## Scope
+
+SIG Apps Covers deploying and operating applications in Kubernetes with a focus on the application developer and application operator experience.
+
+For example, this includes:
+
+* Discussions on all levels from application development, application testing environments, and applications running in production
+* Discussion of how to define and run apps in Kubernetes
+* Demos of relevant tools and projects
+* Discussion on areas of friction that can lead to suggesting improvements or feature requests
+* Development of Kubernetes primitives (e.g., Workloads API) to enable the application developers and operators. Note, most new types should be created as CRDs during their development phase and may always be ecosystem projects
+* Creation of documentation to enable application developers and operators
+* Development and maintenance of the sub-projects of Helm, Charts, and Kompose. Note, these projects are grandfathered and treated as part of the ecosystem but owned by SIG Apps. New projects are expected to stay in the ecosystem.
+
+### Goals
+
+* Discuss running and defining applications in Kubernetes (e.g., APIs, SDKs, Controllers, package management tools, etc.)
+* Share common patterns and practices
+* Work on improvements to the Workload API
+* Suggest and sometimes develop Kubernetes features where we see friction
+* Be the voice of the people running applications in Kubernetes (application developers and operators)
+* Help people get involved in the Kubernetes community
+* Show early features/demos of tools that make running apps easier
+
+### Non-goals
+
+* Do not endorse one particular ecosystem tool
+* Do not pick which apps to run on top of Kubernetes
+* Do not recommend one way to do things (e.g., picking a template language)
+
+## Roles
```

Comment:
You deliberately decided not to suggest any SIG-level technical decision-making entity, either SIG TLs or committee of subproject owners? That would mean, among other things, that the highest escalation point (within the SIG) of a subproject would be the subproject owners. I'll comment on the subproject creation process below.

Reply:
We deviate from the template a little here but let me start with the situation.

* We've not had to mediate between sub-projects at the SIG Apps level.
* SIG Apps covers a vast area. The same technical leads for the Workloads API wouldn't make sense to lead Kompose who wouldn't make sense to lead Helm.
* There have never been tech leads at the SIG level over all the SIG things. It would be artificial to add them now and we would be using top down authority to add a power position rather then trying to fill a need.
* There are currently over 30 sub-project owners for SIG Apps. It's hard to have a committee of that size.

Technical leads over all the things doesn't really make sense here as things sit today. A committee would be hard.

When it comes to mediating between sub-projects:

> Issues impacting multiple subprojects in the SIG should be resolved by SIG Chairs

This is a deviation from the template. I don't expect we'll need to use it. The sub-project owners have always done a great job trying to work together. Any issues were typically outside of SIG Apps and have escalation points elsewhere.

We should revisit this in 6 months and see if it's working. SIG Apps has been a changing thing. When it started it owned no code. Over time that changed to have a number of sub-projects and a lot of activity. I can't predict what will come next. This was laid out because we think it will work for what we have now.

Reply:
I think this makes sense, and actually helps in a sense to foster the diversity of ecosystem.  I'd be ok with revisiting later.

Reply:
If the SIG does not have TLs, who approves the formation or retirement of subprojects?  Its not clear to me in the current charter how a new subproject is accepted by the SIG.

Reply:
ignore above, i see the role is delegated to SIG chairs below.